### PR TITLE
Vuex TypeScript Docs

### PR DESCRIPTION
**Pull request by jefer590**
I noticed that Vuex has already a couple of interfaces to support Typed stores but I didn't found any info about this. So this guide that I made for the Vuex docs is pretty much what i've done in my experience. 

Feedback is very welcome 💪

### CODE REVIEW

File: docs/guide/typescript.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    const { data } = await axios.get('URL')
+    context.commit('mutation', data)
+    await context.dispatch('myotheraction')
+  }
+}
+```
+
+## Mutations
+
+Mutations can be use with the `MutationTree<S>` interface where `S` is the current store type. It works exactly the same as a normal mutation but with the power of types.
```
Comment (ederchrono) [inline docs/guide/typescript.md]:
```suggestion
Mutations can be used with the `MutationTree<S>` interface where `S` is the current store type. It works exactly the same as a normal mutation but with the power of types.
```

File: docs/guide/typescript.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+const mutations: MutationTree<MyModuleState> = {
+  setList (state, newList: string[]): void {
+    state.list = newList // state.list is typed using MyModuleState
+  }
+}
+```
+
+## Getters
+
+getters can be used with the `GetterTree<S, R>` interface which `S` is the current store type and `R` corresponds to the `RootState` and works exactly the same as a normal getter.
```
Comment (ederchrono) [inline docs/guide/typescript.md]:
```suggestion
Getters can be used with the `GetterTree<S, R>` interface which `S` is the current store type and `R` corresponds to the `RootState` and works exactly the same as a normal getter.
```

**Comment by ktsn** [comment]
Thank you for contributing this docs! 
However, we are planing a new way to use Vuex with TypeScript and it will probably be different way with this. And I don't want to officially encourage this way to be honest as it is not type safe way and the most of people uses some wrapper lib to achieve it currently.

**Comment by crutch12** [comment]
@ktsn Hi

Will we ever be able to use Vuex + Typescript without external libs like `vuex-class`?

For now even expression like `store.dispatch('space/action')` returns `any` (because first argument is string)

Do you plan to implement the new way to use of Vuex to support typing?

Thx

**Pull request closed without merging**